addons/pos_self_order/tests/test_self_order_common.py

```diff
@@ -50,3 +50,24 @@ def test_self_order_pos_closed(self):
         for mode in ("mobile", "consultation", "kiosk"):
             self.pos_config.write({"self_ordering_mode": mode})
             self.start_tour(self_route, "self_order_pos_is_closed")
+
+    def test_self_order_preparation_disabling_preparation_display(self):
+        """
+        This test ensures that the preparation display option can be disabled when the self_ordering_mode is set to 'nothing'.
+        It also tests that the preparation display option is enabled automatically when the self_ordering_mode is set to 'kiosk'.
+        """
+        settings = self.env['res.config.settings'].create({'pos_config_id': self.pos_config.id})
+        settings.module_pos_preparation_display = False
+
+        self.pos_config.write({'self_ordering_mode': 'nothing'})
+        settings.pos_config_id = self.pos_config.id
+        settings._onchange_pos_self_order_pay_after()
```

Comment:
Thanks for adding the test @gteboul :+1: 

Got a few comments. Normally we want to avoid calling private methods directly in tests, thus to call onchange in test we usually use `Form`. Any change that you make while using the form should trigger the corresponding onchange. 
If you want examples you can check either of the following two test. They should help you start yours:
* https://github.com/odoo/odoo/blob/f5b41d2e69b6544390d26e482616f47d45f1a2ce/addons/point_of_sale/tests/test_res_config_settings.py#L64
* https://github.com/odoo/odoo/blob/f5b41d2e69b6544390d26e482616f47d45f1a2ce/addons/point_of_sale/tests/test_res_config_settings.py#L36

One last small thing, since the issue came when we have pay after set to `each`, maybe we could start by making sure it is set to that before changing the mode and display.

Don't hesitate to ask me anything if you have any trouble using Form :slightly_smiling_face: Kudos for the double testing !